sql/analyzer/triggers.go

```diff
@@ -497,6 +512,20 @@ func getTriggerLogic(ctx *sql.Context, a *Analyzer, n sql.Node, scope *plan.Scop
 	return triggerLogic, err
 }
 
+// validateNoConflictingColumnNames checks the columns of a joined table to make sure there are no conflicting column
+// names
+func validateNoConflictingColumnNames(n sql.Node) error {
+	sch := n.Schema()
+	columnNames := make(map[string]string)
+	for _, col := range sch {
+		if sourceName, ok := columnNames[col.Name]; ok && sourceName != col.Source {
+			return errors.New("Unable to apply triggers when joined tables have columns with the same name")
+		}
+		columnNames[col.Name] = col.Source
+	}
+	return nil
+}
```

Comment:
I think passing in a schema makes more sense, and if you're checking the conflict before it gets renamed to a scope you can skip the `col.Source` check

```suggestion
func validateNoConflictingColumnNames(sch sql.Schema) error {
	columnNames := make(map[string]struct{})
	for _, col := range sch {
		if _, ok := columnNames[col.Name]; ok {
			return errors.New("Unable to apply triggers when joined tables have columns with the same name")
		}
		columnNames[col.Name] = struct{}{}
	}
	return nil
}
```

Reply:
I check for source because in case a table is joined with its own columns (is that possible?)

Reply:
It's kinda possible?
You need to alias the tables to avoid an error, which just brings us back to case where two different tables have the same column name, but different source.

```
mysql> update a join a set x = 100;
ERROR 1066 (42000): Not unique table/alias: 'a'

mysql> update a as a1 join a as a2 set a1.x = 100;
Query OK, 0 rows affected (0.01 sec)
```

Regardless, we'd still need to be able to differentiate the columns; even if they're from the same table the join makes it so the values in the row are different.